stake-pool/cli/src/main.rs

```diff
@@ -444,50 +567,121 @@ fn command_create_pool(
+    unsafe_fees: bool,
+) -> CommandResult {
+    if !unsafe_fees {
+        check_stake_pool_fees(&epoch_fee, &withdrawal_fee, &deposit_fee)?;
     }
 
+    let reserve_keypair = reserve_keypair.unwrap_or_else(Keypair::new);
+    println!("Creating reserve stake {}", reserve_keypair.pubkey());
+
+    let mint_keypair = mint_keypair.unwrap_or_else(Keypair::new);
+    println!("Creating mint {}", mint_keypair.pubkey());
+
+    let stake_pool_keypair = stake_pool_keypair.unwrap_or_else(Keypair::new);
+
+    let validator_list_keypair = validator_list_keypair.unwrap_or_else(Keypair::new);
+
+    let reserve_stake_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(STAKE_STATE_LEN)?
+        + MINIMUM_RESERVE_LAMPORTS;
+    let mint_account_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(spl_token::state::Mint::LEN)?;
+    let pool_fee_account_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(spl_token::state::Account::LEN)?;
+    let stake_pool_account_lamports = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(get_packed_len::<StakePool>())?;
+    let empty_validator_list = ValidatorList::new(max_validators);
+    let validator_list_size = get_instance_packed_len(&empty_validator_list)?;
+    let validator_list_balance = config
+        .rpc_client
+        .get_minimum_balance_for_rent_exemption(validator_list_size)?;
+    let mut total_rent_free_balances = reserve_stake_balance
+        + mint_account_balance
+        + pool_fee_account_balance
+        + stake_pool_account_lamports
+        + validator_list_balance;
+
+    let default_decimals = spl_token::native_mint::DECIMALS;
+
+    // Calculate withdraw authority used for minting pool tokens
+    let (withdraw_authority, _) = find_withdraw_authority_program_address(
+        &spl_stake_pool::id(),
+        &stake_pool_keypair.pubkey(),
+    );
+
+    if config.verbose {
+        println!("Stake pool withdraw authority {}", withdraw_authority);
+    }
+
+    setup_reserve_stake_account(config, &reserve_keypair, reserve_stake_balance, &withdraw_authority)?;
+    setup_mint_account(config, &mint_keypair, mint_account_balance, &withdraw_authority, default_decimals)?;
+    setup_pool_fee_account(config, &mint_keypair.pubkey(), &mut total_rent_free_balances)?;
+    setup_validator_list_account(config, &validator_list_keypair, validator_list_balance, validator_list_size)?;
+
+    let pool_fee_account = get_associated_token_address(&config.manager.pubkey(), &mint_keypair.pubkey());
+
+    initialize_stake_pool(
+        config,
+        &stake_pool_keypair,
+        &validator_list_keypair,
+        &reserve_keypair,
+        &mint_keypair,
+        &pool_fee_account,
+        deposit_authority,
+        epoch_fee,
+        withdrawal_fee,
+        deposit_fee,
+        referral_fee,
+        max_validators,
+        &withdraw_authority,
+    )?;
```

Comment:
Can we combine `setup_validator_list_account` and `initialize_stake_pool` into the same transaction? Fewer transactions is always best. And although the possibility is really remote, it's technically possible for someone to steal a created-but-not-initialized account, since no data has been written to it yet